cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -128,8 +128,9 @@ data QueryInShelleyBasedEra era result where
 
      QueryStakeAddresses
        :: Set StakeCredential
-       -> QueryInShelleyBasedEra era (Map StakeCredential Lovelace,
-                                      Map StakeCredential PoolId)
+       -> NetworkId
```

Comment:
Why did we need to add the network id? We don't seem to be using it below.

Reply:
We use it here: https://github.com/input-output-hk/cardano-node/pull/2275/files#diff-12e79fbc433fabcfcd46baf0c85e3986c1e0a2460d5a28aac0a5bab5ca3df5ddR446

This allows us to create the `StakeAddress` so we can serialise it more conveniently (vs dealing with a `StakeCredential`)